test/hts-precompile/token-create/tokenCreateCustomContract.js

```diff
@@ -20,79 +20,108 @@
 
 const utils = require('../utils')
 const { expect } = require('chai')
+const { ethers } = require('hardhat')
 const Constants = require('../../constants')
 const { expectValidHash } = require('../assertions')
-const { ethers } = require('hardhat')
 
 describe('TokenCreateCustomContract Test Suite', () => {
-  let signers
   const tokenName = 'WrappedHbar'
   const tokenSymbol = 'WHBAR'
   const tokenMemo = 'Wrapped Hbar'
-  const initialSupply = 1500
-  const maxSupply = 2000
+  const initialSupply = 900000000 // 9 WHBAR
+  const maxSupply = 30000000000 // 300 WHBAR
   const decimals = 8
   const freezeDefaultStatus = false
-  const key = utils.getSignerCompressedPublicKey()
-  let tokenCreateCustomContract
+  let keys, signers, fixedFeeTokenAddress, tokenCreateCustomContract
 
   before(async () => {
-    signers = await ethers.getSigners()
     tokenCreateCustomContract = await utils.deployTokenCreateCustomContract()
-    await utils.updateAccountKeysViaHapi([tokenCreateCustomContract.address])
-  })
+    signers = await ethers.getSigners()
 
-  it('should be able to execute createFungibleTokenPublic with dynamic params', async function () {
-    const tx = await tokenCreateCustomContract.createFungibleTokenPublic(
+    // construct a list of random keys
+    const adminKey = utils.constructIHederaTokenKey(
+      'ADMIN',
+      'SECP256K1',
+      utils.getSignerCompressedPublicKey()
+    )
+
+    const pauseKey = utils.constructIHederaTokenKey(
+      'PAUSE',
+      'SECP256K1',
+      utils.getSignerCompressedPublicKey()
+    )
+
+    const kycKey = utils.constructIHederaTokenKey(
+      'KYC',
+      'CONTRACT_ID',
+      tokenCreateCustomContract.address
+    )
+
+    const supplyKey = utils.constructIHederaTokenKey(
+      'SUPPLY',
+      'CONTRACT_ID',
+      tokenCreateCustomContract.address
+    )
+
+    keys = [adminKey, pauseKey, kycKey, supplyKey]
```

Comment:
Q: So goal here is to have the main EOA be the signer of create, delete and pause transactions. Then have the contract be the authorizer of kyc, mint and burn transactions?
If so why and maybe you should add some comments to make this clear.
Notably wherever you but the EOA as the authorizer you will have to submit a direct call without going through the deployed contract.
It might be easier to start off with just the adminKey set and then in the PRs focused on those scenarios add the necessary key and tests etc